### PR TITLE
added serverside functionality for getting availible students and sho…

### PR DESCRIPTION
**Pull request by Dan2024**
…wing users in a cohort
https://github.com/boolean-uk/team-dev-server/issues/125

### CODE REVIEW

File: src/controllers/user.js [hunk 1]
```diff
@@ -39,14 +39,19 @@ export const getById = async (req, res) => {
 
 export const getAll = async (req, res) => {
   // eslint-disable-next-line camelcase
-  const { first_name: firstName } = req.query
+  const { first_name: firstName, cohort: inCohort } = req.query
+
+  const whereData = {}
+  if (inCohort === 'false') {
+    whereData.cohort = null
```
Comment (Auenc) [inline src/controllers/user.js:46]: Have you tried running the code without this? 

The reason why I'm asking is that, we're defining `whereData` as an empty object, so `whereData.cohort` will be undefined. So it shouldn't limit any users to a specific cohort by default.

Unless the idea here is to search for users who are not in a cohort?

**Review by Auenc — CHANGES_REQUESTED** [review]
Not really a request for changes, more of a clarification.

File: src/controllers/user.js (hunk 1 above)
Comment (Dan2024) [reply src/controllers/user.js:46]: yes the idea is to search for users not in a cohort so we can populate the availible students list we're soon to implement on the client side for https://github.com/boolean-uk/team-dev-client/issues/200

Comment (Auenc) [reply src/controllers/user.js:46]: Okay great! I'll approve this PR then

**Review by Auenc — APPROVED** [review]

**Pull request merged**